File: frontend/src/data/categories.js
// Category из openapi.yaml. code — нижний регистр, как в API (в БД — верхний,
// см. db-schema.md §3.2). Порядок — как в пилюлях категорий на макете.
export const CATEGORIES = [
  { id: 1, code: 'concert', name: 'Концерты', slug: 'concert' },
  { id: 2, code: 'theatre', name: 'Театр', slug: 'theatre' },
  { id: 3, code: 'standup', name: 'Стендап', slug: 'standup' },
  { id: 4, code: 'festival', name: 'Фестивали', slug: 'festival' },
];

export function getCategoryByCode(code) {
  return CATEGORIES.find((category) => category.code === code);
}

File: backend/app/schemas/categories.py
from enum import StrEnum

from pydantic import BaseModel, ConfigDict, Field


class CategoryCode(StrEnum):
    CONCERT = 'concert'
    THEATRE = 'theatre'
    STANDUP = 'standup'
    FESTIVAL = 'festival'

class Category(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: int = Field(..., examples=[1])
    code: CategoryCode = Field(..., 
                    description='Стабильный идентификатор категории для логики приложения. В БД хранится в верхнем регистре (`THEATRE`), в API отдаётся в нижнем.',
                    examples=['theatre']
                               )
    name: str = Field(..., max_length=100, examples=['Театр'])
    slug: str = Field(..., max_length=100, examples=['theatre'])